homeassistant/components/playstation_network/media_player.py

```diff
@@ -0,0 +1,107 @@
+"""Media player entity for the Playstation Network Integration."""
+
+import logging
+
+from homeassistant.components.media_player import (
+    MediaPlayerDeviceClass,
+    MediaPlayerEntity,
+    MediaPlayerEntityDescription,
+    MediaPlayerState,
+    MediaType,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import PlaystationNetworkCoordinator
+from .entity import PlaystationNetworkEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Media Player Entity Setup."""
+    coordinator: PlaystationNetworkCoordinator = config_entry.runtime_data
+
+    if coordinator.data.platform is None:
+        username = coordinator.data.username
+        _LOGGER.warning(
+            "No console found associated with account: %s. -- Skipping creation of media player",
+            username,
+        )
+        return
+
+    async_add_entities([MediaPlayer(coordinator)])
+
+
+class MediaPlayer(PlaystationNetworkEntity, MediaPlayerEntity):
+    """Media player entity representing currently playing game."""
+
+    entity_description = MediaPlayerEntityDescription(
+        key="console",
+        translation_key="console",
+        device_class=MediaPlayerDeviceClass.TV,
+    )
+    _attr_media_image_remotely_accessible = True
+    _attr_translation_key = "playstation"
+    _attr_media_content_type = MediaType.GAME
+
+    def __init__(self, coordinator: PlaystationNetworkCoordinator) -> None:
+        """Initialize PSN MediaPlayer."""
+        super().__init__(coordinator)
+        if coordinator.config_entry:
+            self._attr_unique_id = (
+                f"{coordinator.config_entry.unique_id}_{self.entity_description.key}"
+            )
```

Comment:
put this in the base entity `config_entry: PlaystationNetworkConfigEntry` then you don't need this if, and also perferably use this instead 
```python
if TYPE_CHEKING:
    assert variable_that_should_not_be_none
```